src/eureka/S5_lightcurve_fitting/models/FleckModel.py

```diff
@@ -0,0 +1,217 @@
+
+            # Set all batman parameters
+            for index, item in enumerate(self.longparamlist[chan]):
+                setattr(bm_params, self.paramtitles[index],
+                        self.parameters.dict[item][0])
+                        
+            # Set spot/fleck parameters
+            nspots = 0
+            # set a star rotation for the star object
+            # not actually used in fast mode. 
+            # overwritten if user supplies and runs slow mode
+            star_rotation = 100 
+            fleck_fast = True
+            for key in self.keys:
+                split_key = key.split('_')
+                if len(split_key) == 1:
+                    chan = 0
+                else:
+                    chan = int(split_key[1])
+                if 'rad' in split_key[0]:
+                    # Get the spot number and update self.spotrad
+                    self.spotrad[chan, int(split_key[0][7:])] = \
+                        np.array([self.parameters.dict[key][0]])
+                    nspots += 1
+                elif 'lat' in split_key[0]:
+                    # Get the spot lat and update self.spotlat
+                    self.spotlat[chan, int(split_key[0][7:])] = \
+                        np.array([self.parameters.dict[key][0]])
+                elif 'lon' in split_key[0]:
+                    # Get the spot lon and update self.spotlon
+                    self.spotlon[chan, int(split_key[0][7:])] = \
+                        np.array([self.parameters.dict[key][0]])
+                elif 'con' in split_key[0]:
+                    # Get the spot constrast and assign
+                    spot_contrast = self.parameters.dict[key][0]
+                elif 'rot' in split_key[0]:
+                    # Get the stellar rotation and assign
+                    star_rotation = self.parameters.dict[key][0]
+                    fleck_fast = False
+                elif 'stari' in split_key[0]:
+                    # Get the stellar inclination and assign
+                    star_inc = self.parameters.dict[key][0]
+                elif 'npts' in split_key[0]:
+                    # it's the number of points to evaluate
+                    npoints = self.parameters.dict[key][0]
+            
+            # Set limb darkening parameters
+            uarray = []
+            for u in self.coeffs:
+                index = np.where(np.array(self.paramtitles) == u)[0]
+                if len(index) != 0:
+                    item = self.longparamlist[chan][index[0]]
+                    uarray.append(self.parameters.dict[item][0])
+            bm_params.u = uarray
+
+            # Enforce physicality to avoid crashes from batman by returning
+            # something that should be a horrible fit
+            if not ((0 < bm_params.per) and (0 < bm_params.inc < 90) and
+                    (1 < bm_params.a) and (0 <= bm_params.ecc < 1) and
+                    (0 <= bm_params.w <= 360)):
```

Comment:
We recently removed the constraint on omega (see BatmanModels.py).  You should check for other differences in BatmanModels.py